src/sentry/eventstore/models.py

```diff
@@ -427,8 +430,15 @@ def get_grouping_variants(self, force_config=None, normalize_stacktraces=False):
         return get_grouping_variants_for_event(self, config)
 
     def get_primary_hash(self):
-        flat_hashes, _ = self.get_hashes()
-        return flat_hashes[0] if flat_hashes else None
+        flat_hashes, hierarchical_hashes = self.get_hashes()
+
+        if hierarchical_hashes:
+            return hierarchical_hashes[0]
```

Comment:
Should this not be -1?

Reply:
At this point I am fairly sure that we will need to set the primary hash in Snuba to the root of the tree. The reason is that Merged Issues runs `select distinct primary_hash` to enumerate which hashes it can split by.

If we were to return `hierarchical_hashes[-1]`, you would see all leaf nodes in Merged Issues as a flat list. This can make for a nice intermediate UI (and merge/unmerge will also work), but it causes problems when we want to render something that makes more sense.

Reply:
@mitsuhiko I would definetly roll with this for now, it may be that we have to change it again when we implement merge/unmerge for hierarchical hashes, but I am like 70% sure we don't and for now it should be nbd as long as nothing is working.

Reply:
Let's roll.